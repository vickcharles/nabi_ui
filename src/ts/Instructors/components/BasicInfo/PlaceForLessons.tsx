import * as React from 'react';
import Typography from 'material-ui/Typography';
import Divider from 'material-ui/Divider';
import { FormGroup, FormControlLabel } from 'material-ui/Form';
import Checkbox from 'material-ui/Checkbox';
import TextField from 'material-ui/TextField';

import { PlaceForLessonsOptions } from '../../model';

const PlaceForLessons: React.StatelessComponent<{}> = props => {

  return (
    <div>
      <Typography className="nabi-margin-top-small nabi-margin-bottom-xsmall" variant="body2">
        Place for lessons
      </Typography>
      <FormGroup className="nabi-margin-left-small">
        <FormControlLabel
          control={
            <Checkbox
              // checked={this.state.gilad}
              // onChange={this.handleChange('gilad')}
              value=""
            />
          }
          label={PlaceForLessonsOptions.home}
        />
        <FormControlLabel
          control={
            <Checkbox
              // checked={this.state.jason}
              // onChange={this.handleChange('jason')}
              value=""
            />
          }
          label={PlaceForLessonsOptions.studio}
        />
        <FormControlLabel
          control={
            <Checkbox
              // checked={this.state.antoine}
              // onChange={this.handleChange('antoine')}
              value=""
            />
          }
          label={PlaceForLessonsOptions.online}
        />
      </FormGroup>
      <TextField
        fullWidth={true}
        // id={list.name}
        // name={list.name}
        // onChange={props.handleChange}
        required={true}
        className="nabi-margin-top-xsmall"
        placeholder="Enter your studio address"
        type="number"
        // onBlur={props.updateRates}
      />
      <Divider className="nabi-margin-top-small"/>
    </div>
  );
};

export default (PlaceForLessons);
