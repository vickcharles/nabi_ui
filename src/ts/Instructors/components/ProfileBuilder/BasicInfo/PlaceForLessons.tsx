import * as React from 'react';
import Typography from 'material-ui/Typography';
import Divider from 'material-ui/Divider';
import { FormGroup, FormControlLabel } from 'material-ui/Form';
import Checkbox from 'material-ui/Checkbox';
import TextField from 'material-ui/TextField';

import { placeForLessonsOptions } from '../../../model';

interface PlaceForLessonsProps {
  handleChange: (event: React.FormEvent<{}>) => void;
  handleChangePlaceForLessons: (event: React.FormEvent<{}>) => void;
  home: boolean;
  studio: boolean;
  online: boolean;
  studioAddress: string;
  updateStudioAddress: (event: React.FormEvent<{}>) => void;
}

const PlaceForLessons: React.StatelessComponent<PlaceForLessonsProps> = props => {
  const studioAddressTextfield = props.studio && (
    <TextField
      fullWidth={true}
      name="studioAddress"
      onChange={props.handleChange}
      required={true}
      className="nabi-margin-top-xsmall"
      placeholder="Enter your studio address"
      onBlur={props.updateStudioAddress}
    />
  );

  return (
    <div>
      <Typography className="nabi-margin-top-small nabi-margin-bottom-xsmall" variant="body2">
        Place for lessons
      </Typography>
      <FormGroup className="nabi-margin-left-small">
        <FormControlLabel
          control={
            <Checkbox
              name={placeForLessonsOptions.Home.name}
              checked={props.home}
              onChange={props.handleChangePlaceForLessons}
            />
          }
          label={placeForLessonsOptions.Home.label}
        />
        <FormControlLabel
          control={
            <Checkbox
              name={placeForLessonsOptions.Studio.name}
              checked={props.studio}
              onChange={props.handleChangePlaceForLessons}
            />
          }
          label={placeForLessonsOptions.Studio.label}
        />
        <FormControlLabel
          control={
            <Checkbox
              name={placeForLessonsOptions.Online.name}
              checked={props.online}
              onChange={props.handleChangePlaceForLessons}
            />
          }
          label={placeForLessonsOptions.Online.label}
        />
      </FormGroup>
      {studioAddressTextfield}
      <Divider className="nabi-margin-top-small"/>
    </div>
  );
};

export default (PlaceForLessons);
