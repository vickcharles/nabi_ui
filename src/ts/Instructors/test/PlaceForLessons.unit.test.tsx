import * as React from 'react';
import PlaceForLessons from '../components/BasicInfo/PlaceForLessons';
import { shallow } from 'enzyme';

describe('PlaceForLessons', () => {
  let wrapper: any;

  describe('Shallow', () => {
    beforeEach(() => {
      wrapper = shallow(
        <PlaceForLessons />
      );
    });
    
    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
