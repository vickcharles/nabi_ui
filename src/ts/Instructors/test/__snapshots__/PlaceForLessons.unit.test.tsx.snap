// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceForLessons Shallow Matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlaceForLessons
    handleChange={[MockFunction]}
    handleChangePlaceForLessons={[MockFunction]}
    home={false}
    online={false}
    studio={false}
    studioAddress=""
    updateStudioAddress={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Typography)
          className="nabi-margin-top-small nabi-margin-bottom-xsmall"
          variant="body2"
        >
          Place for lessons
        </WithStyles(Typography)>,
        <WithStyles(FormGroup)
          className="nabi-margin-left-small"
        >
          <WithStyles(FormControlLabel)
            control={
              <WithStyles(Checkbox)
                checked={false}
                name="home"
                onChange={[MockFunction]}
              />
            }
            label="Student's home"
          />
          <WithStyles(FormControlLabel)
            control={
              <WithStyles(Checkbox)
                checked={false}
                name="studio"
                onChange={[MockFunction]}
              />
            }
            label="Instructor's studio"
          />
          <WithStyles(FormControlLabel)
            control={
              <WithStyles(Checkbox)
                checked={false}
                name="online"
                onChange={[MockFunction]}
              />
            }
            label="Online"
          />
        </WithStyles(FormGroup)>,
        false,
        <WithStyles(Divider)
          className="nabi-margin-top-small"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Place for lessons",
          "className": "nabi-margin-top-small nabi-margin-bottom-xsmall",
          "variant": "body2",
        },
        "ref": null,
        "rendered": "Place for lessons",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(FormControlLabel)
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="home"
                  onChange={[MockFunction]}
                />
              }
              label="Student's home"
            />,
            <WithStyles(FormControlLabel)
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="studio"
                  onChange={[MockFunction]}
                />
              }
              label="Instructor's studio"
            />,
            <WithStyles(FormControlLabel)
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="online"
                  onChange={[MockFunction]}
                />
              }
              label="Online"
            />,
          ],
          "className": "nabi-margin-left-small",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "control": <WithStyles(Checkbox)
                checked={false}
                name="home"
                onChange={[MockFunction]}
              />,
              "label": "Student's home",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "control": <WithStyles(Checkbox)
                checked={false}
                name="studio"
                onChange={[MockFunction]}
              />,
              "label": "Instructor's studio",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "control": <WithStyles(Checkbox)
                checked={false}
                name="online"
                onChange={[MockFunction]}
              />,
              "label": "Online",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "nabi-margin-top-small",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            className="nabi-margin-top-small nabi-margin-bottom-xsmall"
            variant="body2"
          >
            Place for lessons
          </WithStyles(Typography)>,
          <WithStyles(FormGroup)
            className="nabi-margin-left-small"
          >
            <WithStyles(FormControlLabel)
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="home"
                  onChange={[MockFunction]}
                />
              }
              label="Student's home"
            />
            <WithStyles(FormControlLabel)
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="studio"
                  onChange={[MockFunction]}
                />
              }
              label="Instructor's studio"
            />
            <WithStyles(FormControlLabel)
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="online"
                  onChange={[MockFunction]}
                />
              }
              label="Online"
            />
          </WithStyles(FormGroup)>,
          false,
          <WithStyles(Divider)
            className="nabi-margin-top-small"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Place for lessons",
            "className": "nabi-margin-top-small nabi-margin-bottom-xsmall",
            "variant": "body2",
          },
          "ref": null,
          "rendered": "Place for lessons",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(FormControlLabel)
                control={
                  <WithStyles(Checkbox)
                    checked={false}
                    name="home"
                    onChange={[MockFunction]}
                  />
                }
                label="Student's home"
              />,
              <WithStyles(FormControlLabel)
                control={
                  <WithStyles(Checkbox)
                    checked={false}
                    name="studio"
                    onChange={[MockFunction]}
                  />
                }
                label="Instructor's studio"
              />,
              <WithStyles(FormControlLabel)
                control={
                  <WithStyles(Checkbox)
                    checked={false}
                    name="online"
                    onChange={[MockFunction]}
                  />
                }
                label="Online"
              />,
            ],
            "className": "nabi-margin-left-small",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "control": <WithStyles(Checkbox)
                  checked={false}
                  name="home"
                  onChange={[MockFunction]}
                />,
                "label": "Student's home",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "control": <WithStyles(Checkbox)
                  checked={false}
                  name="studio"
                  onChange={[MockFunction]}
                />,
                "label": "Instructor's studio",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "control": <WithStyles(Checkbox)
                  checked={false}
                  name="online"
                  onChange={[MockFunction]}
                />,
                "label": "Online",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "nabi-margin-top-small",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
